### PR TITLE
feat(io): add MMCIF support

### LINKED ISSUE #34: MMCIF support
Hi! The PDB format is starting to be moved out of the way in preference of MMCIF. Could I implement some support for this format as its natively supported by TMAlign?

### PR DESCRIPTION
**Pull request by cvigilv**
- **feat(io): add MMCIF support**
- **docs(io): add docstrings**
- **test: cover MMCIF support**

Closes #34

### CODE REVIEW

File: tmtools/io.py [hunk 1]
```diff
@@ -14,13 +15,52 @@
...
 
-def get_structure(fname):
+def get_structure(fname, format="pdb"):
+    """
+    Load and parse a structure file.
+
+    Parameters
+    ----------
+    fname : str
+       Path to the structure file.
```
Comment (jvkersch) [inline tmtools/io.py:25]: One tiny nit, but it seems like this line (and the ones further down in the docstring) are indented by 3 characters, not 4.

File: tmtools/io.py [hunk 2]
```diff
@@ -14,13 +15,52 @@
...
+    Warnings are suppressed during parsing.
+
+    Examples
+    --------
+    >>> structure = get_structure('1abc.pdb')
+    >>> structure = get_structure('1xyz.cif', format='mmcif')
+    """
+    # Check if format provided is valid
+    if format not in ["pdb", "mmcif"]:
+        raise ValueError("Invalid structure format")
```
Comment (jvkersch) [inline tmtools/io.py:51]: Maybe add the format to the error message:

```suggestion
        raise ValueError(f"Invalid structure format: {format!r}")
```

File: tmtools/tests/test_io.py [hunk 3]
```diff
@@ -3,19 +3,23 @@
 import numpy.testing as nptest
 
 from ..io import get_residue_data, get_structure
-from ..testing import get_pdb_path
+from ..testing import get_pdb_path, get_mmcif_path
 
 
 class TestIO(unittest.TestCase):
     def test_get_structure(self):
-        # Given
```
Comment (jvkersch) [inline tmtools/tests/test_io.py:11]: Could you maintain the given/when/then structure here? If those terms are too generic, you can add some explanation, e.g. `# Given - a valid PDB file`, etc.

Also, don't worry about creating a separate test for mmcif, even if it duplicates some code from the pdb test. Likewise if the test ends up checking many things at the same time, that is usually a sign that it should be broken out into multiple tests. Here I would maybe suggest 3 tests: (a) loading PDB, (b) loading MMCIF, (c) checking that the structures are equal.

**Review by jvkersch — CHANGES_REQUESTED** [review]
Couple of stylistic remarks, other than that LGTM. Thanks for doing this, @cvigilv!

File: tmtools/io.py (hunk 1 above)
Comment (cvigilv) [reply tmtools/io.py:25]: You are right! Didn't catch this whenever I was reading this. Thanks for the heads up

File: tmtools/tests/test_io.py (hunk 3 above)
Comment (cvigilv) [reply tmtools/tests/test_io.py:11]: No problem! Please check the changes I did, basically the logic was the following:
- **Given** an structure ID
- **When** we load the structure from PDB/MMCIF
- **Then** we compare both structures to see if this are equal

File: tmtools/io.py (hunk 2 above)
Comment (cvigilv) [reply tmtools/io.py]: Done!

**Comment by cvigilv** [comment]
Should I add some additional test to check expected errors whenever we read a MMCIF file as PDB and viceversa?

**Review by cvigilv — COMMENTED** [review]
Implemented

**Review by jvkersch — APPROVED** [review]
Perfect! Thanks @cvigilv
